api-extractor/src/DocItemLoader.ts

```diff
@@ -49,38 +51,62 @@ export default class DocItemLoader {
   /**
    * {@inheritdoc ApiDocumentation.IReferenceResolver}
    */
-  public resolve(apiDefinitionRef: IApiDefinitionReference, reportError: (message: string) => void): IDocItem {
-    if (!apiDefinitionRef) {
-      reportError('Expected reference within {@inheritdoc} tag');
-      return undefined;
+  public resolve(apiDefinitionRef: IApiDefinitionReference,
+    apiPackage: ApiPackage,
+    reportError: (message: string) => void): ResolvedApiItem {
+
+    const packageName: string = path.dirname(apiPackage.name).split('/').pop();
+
+    let resolvedApiItem: ResolvedApiItem;
+    if (packageName === apiDefinitionRef.packageName || !apiDefinitionRef.packageName) {
+      // Resolution for local references 
+      const apiItem: ApiItem = apiPackage.getMemberItem(apiDefinitionRef.exportName);
+      // Check if export name was not found
+      if (!apiItem) {
+        reportError(`Unable to resolve ApiItem with export name: "${apiDefinitionRef.exportName}"`);
+        return undefined;
+      }
+      // found export name 
+      resolvedApiItem = ResolvedApiItem.createFromApiItem(apiItem);
+    } else {
+       // Resolution for references in JSON files
+      const docPackage: IDocPackage =  this.getPackage(apiDefinitionRef, reportError);
+      // Check if package was not found
+      if (!docPackage) {
+        reportError(`Package "${apiDefinitionRef.packageName}" not found`);
+        return undefined;
+      }
+      // found JSON package, now ensure export name is there 
+      if (!(apiDefinitionRef.exportName in docPackage.exports)) {
+        reportError(`Unable to resolve ApiItem with export name: "${apiDefinitionRef.exportName}"`);
+        return undefined;
+      }
+      const docItem: IDocItem = docPackage.exports[apiDefinitionRef.exportName];
+      resolvedApiItem = ResolvedApiItem.createFromJson(docItem);
     }
-    // Try to load the package given the provided packageName into the cache
-    const docPackage: IDocPackage =  this.getPackage(apiDefinitionRef, reportError);
 
-    // Check if package was not found
-    if (!docPackage) {
-      return undefined;
-    }
+    // We now know the export name has been found and we can use 
+    // resolvedApiItem to abstract the difference between ApiItem and a JSON object (that 
+    // symbolizes an ApiPackage)
 
-    if (apiDefinitionRef.exportName in docPackage.exports) {
-      let docItem: IDocItem = docPackage.exports[apiDefinitionRef.exportName];
+    // If memberName exists then check for the existense of the name
+    if (apiDefinitionRef.memberName) {
+      if (resolvedApiItem.members && apiDefinitionRef.memberName in resolvedApiItem.members) {
+        const memberItem: ApiItem | IDocItem = resolvedApiItem.members[apiDefinitionRef.memberName];
```

Comment:
See my other note.  If you are still doing the lookup to find the final item, you should not be using the ResolvedApiItem abstraction yet.  When you that, you immediately have to unabstract it again on line 97.